sky/serve/load_balancer.py

```diff
@@ -70,30 +74,110 @@ def _sync_with_controller(self):
                     ready_replica_urls = response.json().get(
                         'ready_replica_urls')
                 except requests.RequestException as e:
-                    print(f'An error occurred: {e}')
+                    logger.error(f'An error occurred: {e}')
                 else:
                     logger.info(f'Available Replica URLs: {ready_replica_urls}')
-                    self._load_balancing_policy.set_ready_replicas(
-                        ready_replica_urls)
+                    with self._lock:
```

Comment:
Would be good to have the `_lock` name reflect the purpose of the lock. Also, add a comment for why we have a lock here.

Reply:
Added! Thanks.